File: memory-domain/src/main/java/com/memory/dto/comment/CommentUpdateRequest.java
package com.memory.dto.comment;

import jakarta.validation.constraints.NotBlank;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@AllArgsConstructor
public class CommentUpdateRequest {

    @NotBlank(message = "댓글 내용은 필수입니다.")
    private String content;
}

File: memory-api/src/main/kotlin/com/memory/dto/comment/CommentUpdateRequest.kt
package com.memory.dto.comment

import jakarta.validation.constraints.NotBlank

data class CommentUpdateRequest private constructor(
    @field:NotBlank(message = "댓글 내용은 필수입니다.")
    val content: String,
) {
    companion object {
        @JvmStatic
        fun of(content: String): CommentUpdateRequest =
            CommentUpdateRequest(content = content)

        @JvmStatic
        fun testInstance(content: String): CommentUpdateRequest =
            CommentUpdateRequest(content = content)
    }
}